driver-opentelemetry/src/test/java/com/datastax/driver/opentelemetry/OpenTelemetryTest.java

```diff
@@ -0,0 +1,165 @@
+   * @return collected spans.
+   */
+  private Collection<ReadableSpan> collectSpans(BiConsumer<Tracer, TracingInfoFactory> test) {
+    final Resource serviceNameResource =
+        Resource.create(
+            Attributes.of(ResourceAttributes.SERVICE_NAME, "Scylla Java driver - test"));
+
+    final SpansCollector collector = new SpansCollector();
+
+    final SdkTracerProvider tracerProvider =
+        SdkTracerProvider.builder()
+            .addSpanProcessor(collector)
+            .setResource(Resource.getDefault().merge(serviceNameResource))
+            .build();
+    final OpenTelemetrySdk openTelemetry =
+        OpenTelemetrySdk.builder().setTracerProvider(tracerProvider).buildAndRegisterGlobal();
+
+    final Tracer tracer = openTelemetry.getTracerProvider().get("this");
+    final OpenTelemetryTracingInfoFactory tracingInfoFactory =
+        new OpenTelemetryTracingInfoFactory(tracer);
+    session().setTracingInfoFactory(tracingInfoFactory);
+
+    test.accept(tracer, tracingInfoFactory);
+
+    tracerProvider.close();
+    session().setTracingInfoFactory(new NoopTracingInfoFactory());
+
+    return collector.getSpans();
+  }
+
+  @Override
+  public void onTestContextInitialized() {
+    execute("CREATE TABLE t (k int PRIMARY KEY, v int)");
+  }
+
+  /** Basic test for creating spans. */
+  @Test(groups = "short")
+  public void simpleTracingTest() {
+    final Collection<ReadableSpan> spans =
+        collectSpans(
+            (tracer, tracingInfoFactory) -> {
+              Span userSpan = tracer.spanBuilder("user span").startSpan();
+              Scope scope = userSpan.makeCurrent();
+
+              session().execute("INSERT INTO t(k, v) VALUES (4, 2)");
+              session().execute("INSERT INTO t(k, v) VALUES (2, 1)");
+
+              scope.close();
+              userSpan.end();
+            });
+
+    // Retrieve span created directly by tracer.
+    final List<ReadableSpan> userSpans =
+        spans.stream()
+            .filter(span -> !span.getParentSpanContext().isValid())
+            .collect(Collectors.toList());
+    assertEquals(userSpans.size(), 1);
+    final ReadableSpan userSpan = userSpans.get(0);
+
+    for (ReadableSpan span : spans) {
+      assertTrue(span.getSpanContext().isValid());
+      assertTrue(
+          span.getSpanContext().equals(userSpan.getSpanContext())
+              || span.getParentSpanContext().isValid());
+    }
+
+    // Retrieve spans representing requests.
+    final Collection<ReadableSpan> rootSpans =
+        spans.stream()
+            .filter(span -> span.getParentSpanContext().equals(userSpan.getSpanContext()))
+            .collect(Collectors.toList());
+    assertEquals(rootSpans.size(), 2);
+
+    rootSpans.stream()
+        .map(ReadableSpan::toSpanData)
+        .forEach(
+            spanData -> {
+              assertEquals(spanData.getName(), "request");
+              assertEquals(spanData.getStatus().getStatusCode(), StatusCode.OK);
```

Comment:
Maybe you should put more `assertEquals` for other OT tags (I know it might be dependant on PR #5).